src/test/java/it/com/atlassian/bitbucket/jenkins/internal/status/BuildStatusPosterIT.java

```diff
@@ -0,0 +1,300 @@
+                PROJECT_KEY, repoSlug, gitHelper.getLatestCommit());
+        bitbucketProxyRule.getWireMock().stubFor(post(
+                urlPathMatching(url))
+                .willReturn(aResponse().withStatus(HttpStatus.SC_NO_CONTENT)));
+
+        jenkinsProjectHandler.runPipelineJob(job, build -> {
+            try {
+                verify(postRequestedFor(urlPathMatching(url))
+                        .withRequestBody(
+                                equalToJson("{" +
+                                            format("\"description\":\"%s successful in %s\",",
+                                                    build.getDisplayName(), build.getDurationString()) +
+                                            format("\"duration\":%d,", build.getDuration()) +
+                                            format("\"key\":\"%s\",", job.getName()) +
+                                            format("\"name\":\"%s\",", job.getName()) +
+                                            //TODO: verify the ref: "ref" : "refs/heads/rep-1-fork/master"
+                                            format("\"resultKey\":\"%s\",", build.getExternalizableId()) +
+                                            format("\"serverIdentifier\":\"%s\",", bbJenkinsRule.getURL()) +
+                                            format("\"url\":\"%sjob/%s/%s/display/redirect\",",
+                                                    bbJenkinsRule.getURL(), job.getName(), build.getId()) +
+                                            format("\"state\":\"%s\"", "SUCCESSFUL") +
+                                            "}", true, true))
+                );
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        });
+    }
+
+    @Test
+    public void testAgainstPipelineWithBitbucketSCM() throws Exception {
+        WorkflowJob wfj =
+                jenkinsProjectHandler.createPipelineJobWithBitbucketScm("wfj", repoSlug, MASTER_BRANCH_PATTERN);
+
+        String latestCommit = checkInJenkinsFile(
+                "pipeline {\n" +
+                "    agent any\n" +
+                "\n" +
+                "    stages {\n" +
+                "        stage('Build') {\n" +
+                "            steps {\n" +
+                "                echo 'Building..'\n" +
+                "            }\n" +
+                "        }\n" +
+                "    }\n" +
+                "}");
+
+        String url =
+                format("/rest/api/1.0/projects/%s/repos/%s/commits/%s/builds/([a-z0-9]*)", PROJECT_KEY, repoSlug, latestCommit);
+        bitbucketProxyRule.getWireMock().stubFor(post(
+                urlPathMatching(url))
+                .willReturn(aResponse().withStatus(HttpStatus.SC_NO_CONTENT)));
+
+        jenkinsProjectHandler.runPipelineJob(wfj);
+
+        verify(postRequestedFor(urlPathMatching(url))
+                //TODO
+//                .withRequestBody(
```

Comment:
No thrilled about lots of commented out code, do we plan on addressing this TODO before merging? If not I think we should remove this block.